applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c

```diff
@@ -386,10 +417,67 @@ static int device_index_get(const struct bt_conn *conn, struct stream_index *idx
 	return -EINVAL;
 }
 
-static int device_index_vacant_get(const struct bt_conn *conn, struct stream_index *idx)
+/**
+ * @brief	Get device index based on stream pointer.
+ *
+ * @param[in]	stream	The stream to search for.
+ * @param[out]	index	The device index.
+ *
+ * @retval	0	Operation successful.
+ * @retval	-EINVAL	There is no match.
+ */
+static int device_index_from_stream_get(const struct bt_bap_stream *stream,
+					struct stream_index *idx)
+{
+	if (stream == NULL) {
+		LOG_ERR("No stream provided");
+		return -EINVAL;
+	}
+
+	if (idx == NULL) {
+		LOG_ERR("No index provided");
+		return -EINVAL;
+	}
+
+	for (int i = 0; i < CONFIG_BT_ISO_MAX_CIG; i++) {
+		for (int j = 0; j < ARRAY_SIZE(unicast_servers[i][0]); j++) {
+			if (&unicast_servers[i][0][j].cap_sink_stream.bap_stream == stream ||
+			    &unicast_servers[i][0][j].cap_source_stream.bap_stream == stream) {
+				idx->lvl1 = i;
+				idx->lvl2 = 0;
```

Comment:
IIRC the lvl2 index is only used for broadcast and represent the subgroups, right? 

Consider defining a value like `#define STREAM_INDEX_UNICAST_LVL2 0` or something that be used instead of the literal `0` in cases where you want to access the lvl2 for unicast